static/app/utils/discover/discoverOrMetricsQuery.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  convertDiscoverFieldsToMetrics,
+  convertDiscoverFieldsToMetricsGroupBys,
+} from 'sentry/utils/metrics/fields';
+import MetricsRequest from 'sentry/utils/metrics/metricsRequest';
+import {useMetricsSwitch} from 'sentry/views/performance/metricsSwitch';
+
+import withApi from '../withApi';
+
+import DiscoverQuery, {DiscoverQueryPropsWithThresholds} from './discoverQuery';
+
+function DiscoverOrMetricsQuery(props: DiscoverQueryPropsWithThresholds) {
+  const {isMetricsData} = useMetricsSwitch(); // TODO(metrics): decision logic will be dynamic in the future, not switch toggle
+
+  if (isMetricsData) {
+    const {eventView, orgSlug, api} = props;
+    const {start, end, statsPeriod, project, environment} = eventView;
+    const fields = eventView.fields.map(({field}) => field);
+
+    return (
+      <MetricsRequest
+        api={api}
+        orgSlug={orgSlug}
+        start={start}
+        end={end}
+        statsPeriod={statsPeriod}
+        project={project}
+        environment={environment}
+        // query={eventView.getQueryWithAdditionalConditions()} TODO(metrics)
+        field={convertDiscoverFieldsToMetrics(fields)}
+        groupBy={convertDiscoverFieldsToMetricsGroupBys(fields)}
+        orderBy={undefined} // TODO(metrics): waiting for api team_key_transactions sorting
```

Comment:
Even without `team_key_transactions` working with metrics, can we order by the secondary column selected for sorting here? Then we can see the pagination work done on the backend in action.

Reply:
I'll tacke both sorting and pagination in a follow-up PR. Tracking it here in JIRA: 
https://getsentry.atlassian.net/browse/INGEST-871
https://getsentry.atlassian.net/browse/INGEST-872

Reply:
OK perfect!